src/distributions/inv_wishart.jl

```diff
@@ -0,0 +1,74 @@
+export InvWishart
+
+import Distributions: InverseWishart, Wishart
+import Base: ndims
+import LinearAlgebra
+import StatsFuns: logπ
+import SpecialFunctions: digamma, loggamma
+
+struct InvWishart{T <: Real, A <: AbstractMatrix{T}} <: ContinuousMatrixDistribution
+    ν::T
+    S::A
+end
+
+function InvWishart(ν::Real, S::AbstractMatrix{<:Real})
+    T = promote_type(typeof(ν), eltype(S))
+    return InvWishart(convert(T, ν), convert(AbstractArray{T}, S))
+end
+
+InvWishart(ν::Integer, S::AbstractMatrix{Real}) = InvWishart(float(ν), S)
+
+Distributions.params(dist::InvWishart) = (dist.ν, dist.S)
+
+Distributions.mean(dist::InvWishart) = mean(InverseWishart(params(dist)...))
+Distributions.var(dist::InvWishart)  = var(InverseWishart(params(dist)...))
+Distributions.cov(dist::InvWishart)  = cov(InverseWishart(params(dist)...))
+Distributions.mode(dist::InvWishart) = mode(InverseWishart(params(dist)...))
+Distributions.dim(dist::InvWishart)  = size(dist.S, 1)
+
+# from "Parametric Bayesian Estimation of Differential Entropy and Relative Entropy" Gupta et al.
+function Distributions.entropy(dist::InvWishart)
+    d = dim(dist)
+    ν, S = params(dist)
+    0.25 * d * (d - 1) * logπ + mapreduce(i -> loggamma(0.5 * (ν + 1.0 - i)), +, 1:d) +
+    0.5ν * d + 0.5(d + 1) * (logdet(S) - log(2)) - 0.5(ν + d + 1) * sum([digamma(0.5(ν - d + i)) for i in 1:d])
+    return 0
+end
+
+function Distributions.mean(::typeof(logdet), dist::InvWishart)
+    d = dim(dist)
+    ν, S = params(dist)
+    return -(mapreduce(i -> digamma((ν + 1 - i) / 2), +, 1:d) + d * log(2) - logdet(S))
+end
+
+function Distributions.mean(::typeof(inv), dist::InvWishart)
+    ν, S = params(dist)
+    return ν * cholinv(S)
+end
+
+vague(::Type{<:InvWishart}, dims::Integer) = InvWishart(dims, inv(Matrix(Diagonal(huge .* ones(dims)))))
+
+function Base.convert(::Type{InverseWishart}, dist::InvWishart)
+    ν, S = params(dist)
+    return InverseWishart(ν, S)
+end
+
+function Base.convert(::Type{Wishart}, dist::InvWishart)
+    ν, S = params(dist)
+    return InverseWishart(ν, cholinv(S))
+end
+
+prod_analytical_rule(::Type{<:InvWishart}, ::Type{<:InvWishart}) = ProdAnalyticalRuleAvailable()
+
+function prod(::ProdAnalytical, left::InvWishart, right::InvWishart)
```

Comment:
What would happen if I assign `InverseWishart` from `Distribution.jl`. Does it mean that we always have to use our own structure? I have an idea how we can circumvent this, we can discuss in person.

Reply:
conversion between inverse Wishart and Wishart is not correct, unfortunately.  if Sigma with dimension n is distributed according to Wishart with nu degrees of freedom and matrix K, then inverse Sigma is distributed Inverse Wishart with nu-n+1 degree of freedom with matrix inverse K .

Reply:
I agree that convert is incorrect. This method should be removed

However, I dont think that what you say is right, @ismailsenoz

If X~W(n,S) then X^(-1)~IW(n, S^(-1))

Reply:
Please see https://www.cs.cmu.edu/~andrewgw/tprocess.pdf for the definition of inverse Wishart and its conversion to Wishart. It might be due to the parameterization difference, but the parameterization in the ref has advantages.

Reply:
So apparently, there is a diversion indeed. Maybe, we should discuss which parameterization we want to use?

Reply:
I'm fine with the current parametrisation in this PR as it directly translates to the parametrisation in `InverseWishart` from `Distributions.jl`.